local-cli/generator-android/templates/src/app/proguard-rules.pro

```diff
@@ -40,10 +40,13 @@
 
 -keep class * extends com.facebook.react.bridge.JavaScriptModule { *; }
 -keep class * extends com.facebook.react.bridge.NativeModule { *; }
+-keep class * { native <methods>; }
```

Comment:
Can you try changing this to?

    # For native methods, see http://proguard.sourceforge.net/manual/examples.html#native
    -keepclasseswithmembernames class * {
        native <methods>;
    }

`keep class` gives 7.1MB APK, `keepclasseswithmembernames` 6.8MB.